providers/ses/src/lib/ses.provider.ts

```diff
@@ -56,13 +74,34 @@ export class SESEmailProvider implements IEmailProvider {
     };
   }
 
-  async checkIntegration(
-    options: IEmailOptions
-  ): Promise<ICheckIntegrationResponse> {
+  async checkIntegration(): Promise<ICheckIntegrationResponse> {
+    let success: boolean;
+    let message: string;
+    let code: CheckIntegrationResponseEnum;
+
+    const testResponse = await this.sendMail({
+      html: '',
+      text: 'This is a Test mail to test your Amazon SES integration',
+      to: this.config.from,
+      from: this.config.from,
+      subject: 'Test SES integration',
+      attachments: {},
+    });
+
+    if (testResponse.err === null) {
+      success = true;
+      message = 'Integration test was succesful';
+      code = CheckIntegrationResponseEnum.SUCCESS;
+    } else {
+      success = false;
+      message = testResponse;
+      code = CheckIntegrationResponseEnum.FAILED;
+    }
+
```

Comment:
```suggestion
  async checkIntegration(): Promise<ICheckIntegrationResponse> {
   try {
    await this.sendMail({
      html: "",
      text: "This is a Test mail to test your Amazon SES integration",
      to: this.config.from,
      from: this.config.from,
      subject: "Test SES integration",
      attachments: {},
    });

    return {
      success: true,
      message: "Integrated Successfully",
      code: CheckIntegrationResponseEnum.SUCCESS,
    };
  } catch (error) {
    return {
      succes: false,
      message: error?.message,
      code: CheckIntegrationResponseEnum.FAILED,
    };
  }
 }

```
Update the `checkIntegration` method to use trycatch something like this